backend/ipmi.pm

```diff
@@ -142,6 +149,57 @@ sub get_mc_status {
     $self->ipmitool("mc selftest");
 }
 
+sub do_mc_reset {
+    my ($self) = @_;
+
+    # deactivate sol console before doing mc reset because it breaks sol connection
+    if (defined $testapi::distri->{consoles}->{'sol'}) {
+        bmwqemu::diag("Before doing mc reset, sol console exists, so cleanup it");
+        $testapi::distri->{consoles}->{'sol'}->reset();
+        bmwqemu::diag("sol console reset done");
+        $self->deactivate_console({testapi_console => 'sol'});
+        bmwqemu::diag("deactivate console sol done");
+    }
+
+    # during the eval execution of following commands, SIG{__DIE__} will definitely be triggered, let it go
```

Comment:
why? the eval is supposed to catch anyway

Reply:
This class inherits the baseclass's signal handler for DIE, and will eventually trigger the die operation without this change, although when $@ catched we have a if branch to handle.